frontend/src/features/public-form/PublicFormProvider.tsx

```diff
@@ -48,6 +48,16 @@ interface PublicFormProviderProps {
 }
 
 export function useCommonFormProvider(formId: string) {
+  // For mobile section sidebar
+  const [isMobileSectionSidebarOpen, setIsMobileSectionSidebarOpen] =
+    useState<boolean>(false)
+  const handleMobileSectionSidebarOpen = useCallback(() => {
+    setIsMobileSectionSidebarOpen(true)
+  }, [setIsMobileSectionSidebarOpen])
+  const handleMobileSectionSidebarClose = useCallback(() => {
+    setIsMobileSectionSidebarOpen(false)
+  }, [setIsMobileSectionSidebarOpen])
```

Comment:
/nit: it is not necessary to specify `setIsMobileSectionSidebarOpen` in the dependency array of `useCallBack()`. `setIsMobileSectionSidebarOpen` is a setter generated by `useState()`, which guarantees it is constant.

From [documentation](https://reactjs.org/docs/hooks-reference.html#usestate):

> React guarantees that setState function identity is stable and won’t change on re-renders. This is why it’s safe to omit from the useEffect or useCallback dependency list.